### PR TITLE
Add .NET, Kotlin, Java function examples to Functions Guide

### PR DESCRIPTION
**Pull request by adityaoberai**
Adds a .NET, Kotlin, and Java function examples to the [Writing your own Function](https://appwrite.io/docs/functions#writingYourOwnFunction) section in the Functions guide.

### CODE REVIEW

**Comment by Meldiron** [comment]
Screenshots:

CleanShot 2022-06-07 at 12 32 05

CleanShot 2022-06-07 at 12 32 01

CleanShot 2022-06-07 at 12 31 54

File: app/views/docs/functions.phtml [hunk 1]
```diff
@@ -251,6 +251,106 @@ Future &lt;void&gt; start(final req, final res) async {
...
+    <li>
+        <h3>Kotlin</h3>
+        <div>
+        <div class="ide margin-top-small" data-lang="kotlin" data-lang-label="Kotlin">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import kotlin.random.Random
+
+@Throws(Exception::class)
+fun main(req: RuntimeRequest, res: RuntimeResponse): RuntimeResponse {
+
+    val payload = if (req.payload.isEmpty() || req.payload == "{}") "No payload provided. Add custom data when executing function." else req.payload
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:295]:
```suggestion
    val payload = if (req.payload.isEmpty()) "No payload provided. Add custom data when executing function." else req.payload
```

We no longer need `'{}'` check. This is no longer a default.

File: app/views/docs/functions.phtml [hunk 2]
```diff
@@ -251,6 +251,106 @@ Future &lt;void&gt; start(final req, final res) async {
...
+
+    val secretKey = req.env["SECRET_KEY"] ?: "SECRET_KEY environment variable not found. You can set it in Function settings."
+    
+    val randomNumber = Random.nextInt(0, 30)
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:299]: Let's make this random from 0 to 1, to keep it consistent with other examples.

File: app/views/docs/functions.phtml [hunk 3]
```diff
@@ -251,6 +251,106 @@ Future &lt;void&gt; start(final req, final res) async {
...
+        "message" to "Hello from Appwrite!",
+        "payload" to payload,
+        "secretKey" to secretKey,
+        "randomNumber" to randomNumber,
+        "trigger" to trigger
+    ))
+}</code></pre>
+        </div>
+        <p><b>Installing Dependencies</b></p>
+        <p>Include a <b>deps.gradle</b> file in your function, Appwrite handles the rest!</p>
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:313]: What about Maven? Do we support it, if so, can we document it? If not, why? Let's talk to Bradley if we need to support it.

File: app/views/docs/functions.phtml [hunk 4]
```diff
@@ -251,6 +251,106 @@ Future &lt;void&gt; start(final req, final res) async {
...
+                     ? "No payload provided. Add custom data when executing function."
+                     : req.getPayload();
+
+    Map<String, String> env = req.getEnv();
+
+    String secretKey = env.containsKey("SECRET_KEY")
+                       ? env.get("SECRET_KEY")
+                       : "SECRET_KEY environment variable not found. You can set it in Function settings.";
+
+    int randomNumber = new Random().nextInt();
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:336]: Can we make this float (or double) number between 0 and 1?

File: app/views/docs/functions.phtml [hunk 5]
```diff
@@ -251,6 +251,106 @@ Future &lt;void&gt; start(final req, final res) async {
...
+        response.put("payload", payload);
+        response.put("secretKey", secretKey);
+        response.put("randomNumber", randomNumber);
+        response.put("trigger", trigger);
+
+    return res.json(response);
+}</code></pre>
+        </div>
+        <p><b>Installing Dependencies</b></p>
+        <p>Include a <b>deps.gradle</b> file in your function, Appwrite handles the rest!</p>
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:351]: Same comment as previously regarding Maven.

**Review by Meldiron — CHANGES_REQUESTED** [review]
After applying these, please double-check if they still work. We might have a bug in open-runtimes if these changes don't work. (payload should always default to empty string)

File: app/views/docs/functions.phtml [hunk 6]
```diff
@@ -251,6 +251,106 @@ Future &lt;void&gt; start(final req, final res) async {
         <p>Include a <b>Package.swift</b> file with your function, Appwrite handles the rest!</p>
         </div>
     </li>
+    <li>
+        <h3>.NET</h3>
+        <div>
+        <div class="ide margin-top-small" data-lang="csharp" data-lang-label="C#">
+            <pre class="line-numbers"><code class="prism language-csharp" data-prism>public async Task<RuntimeResponse> Main(RuntimeRequest req, RuntimeResponse res)
+{
+  var payload = (string.IsNullOrEmpty(req.Payload) || req.Payload.Equals("{}"))
```
[the remark this answers is not in the document]
Comment (adityaoberai) [reply app/views/docs/functions.phtml:260]: No, we don't. 

Also, I was getting a `{}` here however when I sent an empty input from the console.
image

File: app/views/docs/functions.phtml (hunk 4 above)
Comment (adityaoberai) [reply app/views/docs/functions.phtml:336]: Will change that

**Comment by gewenyu99** [comment]
I think this looks alright, I have no comments on the code itself, mainly because I don't know kotlin or C# well. I would think adding some comments might be useful.

It depends on the language, but we should at least comment on what the function is supposed to do in a comment.

**Review by gewenyu99 — COMMENTED** [review]
I would suggest adding some comments so the code is easier to follow. I say this mostly because I don't know these languages very well, and I subconsciously started looking for a comment.

**Comment by adityaoberai** [comment]
> I would suggest adding some comments so the code is easier to follow. I say this mostly because I don't know these languages very well, and I subconsciously started looking for a comment.

That's actually not been across any of the samples. Maybe something we should have across them all?

**Comment by adityaoberai** [comment]
@Meldiron I removed the check for `{}` with the check for a null payload and got this in each case:

* .NET

image

* Java

**Review by Meldiron — APPROVED** [review]
LGTM

**Comment by Meldiron** [comment]
@adityaoberai don't worry about `'{}'` payload, it is a bug in Appwrite. Addressed in this PR: https://github.com/appwrite/appwrite/pull/3367

Payload (should, and) will be empty string if not provided.

**Comment by adityaoberai** [comment]
Sure 👍 
I have updated the function examples to check the payload for an empty string only.

**Comment by adityaoberai** [comment]
> I think this looks alright, I have no comments on the code itself, mainly because I don't know kotlin or C# well. I would think adding some comments might be useful.
> 
> It depends on the language, but we should at least comment on what the function is supposed to do in a comment.

@gewenyu99 maybe this could be an entirely separate issue of it's own that we could work on? I'll go ahead and create one in the meantime.

File: app/views/docs/functions.phtml [hunk 7]
```diff
@@ -251,6 +251,105 @@ Future &lt;void&gt; start(final req, final res) async {
...
+        "message" to "Hello from Appwrite!",
+        "payload" to payload,
+        "secretKey" to secretKey,
+        "randomNumber" to randomNumber,
+        "trigger" to trigger
+    ))
+}</code></pre>
+        </div>
+        <p><b>Installing Dependencies</b></p>
+        <p>Include a <b>deps.gradle</b> file in your function, Appwrite handles the rest!</p>
```
Comment (christyjacob4) [inline app/views/docs/functions.phtml:313]:
```suggestion
        <p>Include a <b>deps.gradle</b> file with your function, Appwrite handles the rest!</p>
```
Likewise

File: app/views/docs/functions.phtml [hunk 8]
```diff
@@ -251,6 +251,105 @@ Future &lt;void&gt; start(final req, final res) async {
...
+    { "message", "Hello from Appwrite!" },
+    { "payload", payload },
+    { "secretKey", secretKey },
+    { "randomNumber", randomNumber },
+    { "trigger", trigger },
+  });
+}</code></pre>
+        </div>
+        <p><b>Installing Dependencies</b></p>
+        <p>Include a <b>Function.csproj</b> file in your function, Appwrite handles the rest!</p>
```
[the remark this answers is not in the document]
Comment (adityaoberai) [reply app/views/docs/functions.phtml:283]: Will do that 👍

File: app/views/docs/functions.phtml [hunk 9]
```diff
@@ -251,6 +251,105 @@ Future &lt;void&gt; start(final req, final res) async {
...
+        response.put("payload", payload);
+        response.put("secretKey", secretKey);
+        response.put("randomNumber", randomNumber);
+        response.put("trigger", trigger);
+
+    return res.json(response);
+}</code></pre>
+        </div>
+        <p><b>Installing Dependencies</b></p>
+        <p>Include a <b>deps.gradle</b> file in your function, Appwrite handles the rest!</p>
```
Comment (christyjacob4) [inline app/views/docs/functions.phtml:350]:
```suggestion
        <p>Include a <b>deps.gradle</b> file in your function - Appwrite handles the rest!</p>
```

**Comment by adityaoberai** [comment]
@christyjacob4 made the grammatical (+ a couple of other minor ones I saw)

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**